lib/main.dart

```diff
@@ -0,0 +1,271 @@
+      ),
+      builder: BotToastInit(),
+      navigatorObservers: [BotToastNavigatorObserver()],
+      home: const MyHomePage(title: 'Take Home Project'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  List<ProductItem>? listProduct = [];
+  bool isFirstLoad = true;
+
+  Future fetchApi() async {
+    final url = Uri.parse('https://hf-android-app.s3-eu-west-1.amazonaws.com/android-test/recipes.json');
+    try {
+      final response = await http.get(url);
+      if(response.statusCode == 200) {
+        final extractedData = jsonDecode(response.body);
+        for (final data in extractedData) {
+          listProduct?.add(ProductItem.fromJson(data));
+        }
+      }
+    } catch (e) {
+      BotToast.showSimpleNotification(
+          title: 'Error fetching data',
+          titleStyle: const TextStyle(color: CupertinoColors.white),
+          duration: const Duration(seconds: 2),
+          align: Alignment.bottomCenter,
+          backgroundColor: Colors.red);
+    }
+    isFirstLoad = false;
+    setState(() {});
+  }
+
+  String parseISODuration(String isoString) {
+    String timePart = isoString.replaceFirst('PT', '');
+
+    int hours = 0;
+    int minutes = 0;
+    int seconds = 0;
+    String time = '';
+
+    RegExp exp = RegExp(r'(\d+H)?(\d+M)?(\d+S)?');
+    RegExpMatch? match = exp.firstMatch(timePart);
+
+    if (match != null) {
+      if (match.group(1) != null) {
+        hours = int.parse(match.group(1)!.replaceFirst('H', ''));
+        time += '${hours}H';
+      }
+      if (match.group(2) != null) {
+        minutes = int.parse(match.group(2)!.replaceFirst('M', ''));
+        time += '${minutes}M';
+      }
+      if (match.group(3) != null) {
+        seconds = int.parse(match.group(3)!.replaceFirst('S', ''));
+        time += '${seconds}S';
+      }
+    }
+    return time;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    fetchApi();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          backgroundColor: Theme.of(context).colorScheme.inversePrimary,
+          title: Text(widget.title),
+          elevation: 0,
+        ),
+        body: isFirstLoad
+            ? Center(
+                child: CircularProgressIndicator(
+                  color: Theme.of(context).colorScheme.inversePrimary,
+                ),
+              )
+            : listProduct?.isEmpty ?? false
+                ? buildEmptyData(title: 'No data found')
+                : ListView.separated(
+                    physics: const BouncingScrollPhysics(),
+                    padding: const EdgeInsets.all(16),
+                    itemBuilder: (context, index) {
+                      return buildItem(productItem: listProduct?[index]);
```

Comment:
Avoid using function over class create a reusable widget.
- [Widgets vs helper methods | Decoding Flutter](https://www.youtube.com/watch?v=IOyq-eTRhvo)
- [More](https://medium.com/@abdurrehman-520/functions-vs-classes-to-create-reusable-widgets-in-flutter-2cf0678ea39a)